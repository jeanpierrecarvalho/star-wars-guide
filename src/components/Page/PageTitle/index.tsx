interface IProps {
  title: string;
}

const PageTitle: React.FC<IProps> = ({ title }) => {
  return (
    <div className="border-2 border-yellow-500 bg-gray-900 px-2 pb-2 border-opacity-60">
      <h1
        className="font-starjout text-5xl text-center text-yellow-500"
        role="heading"
        aria-level={1}
        aria-label={`Page title: ${title}`}
        id="page-title"
      >
        {title}
      </h1>
    </div>
  );
};

export default PageTitle;
