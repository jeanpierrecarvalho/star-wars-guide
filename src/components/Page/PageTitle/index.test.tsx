import { render } from '@testing-library/react';
import { vi, describe, it, expect } from 'vitest';

import PageTitle from '@/components/Page/PageTitle';

vi.mock('next/image', () => ({
	_esModule: true,
	default: ({ src, alt }: { src: string; alt: string }) => {
		return <img src={src} alt={alt} />;
	}
}));

describe('PageTitle Component', () => {
	it('matches the snapshot', () => {
		const { container } = render(<PageTitle title={'Example'} />);
		expect(container).toMatchSnapshot();
	});
});
