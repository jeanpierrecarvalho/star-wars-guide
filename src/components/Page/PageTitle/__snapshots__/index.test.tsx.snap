// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PageTitle Component > matches the snapshot 1`] = `
<div>
  <h1
    class="font-starjout text-5xl text-center text-yellow-500"
  >
    Example
  </h1>
</div>
`;
